fondant/component.py

```diff
@@ -92,15 +92,19 @@ class FondantLoadComponent(FondantComponent):
     """Abstract base class for a Fondant load component"""
 
     def _load_or_create_manifest(self) -> Manifest:
-        metadata = json.loads(self.args.metadata)
+        # create initial manifest
         # TODO ideally get rid of args.metadata by including them in the storage args, getting
         #  run_id based on args.output_manifest_path
+        metadata = json.loads(self.args.metadata)
         manifest = Manifest.create(
             base_path=metadata["base_path"],
             run_id=metadata["run_id"],
             component_id=metadata["component_id"],
         )
 
+        # evolve manifest based on component spec
+        manifest = manifest.evolve(self.spec)
```

Comment:
I don't think we need to evolve the manifest here already, as we need the input manifest first to load the data.

Reply:
This is a loading component, so there's no data to be loaded here. For a transform component we can indeed only evolve after we have provided the dataframe to the user (see below)

Reply:
Ah ok, makes sense.

I think it still might be more logical to always return the input manifest from this method though, and evolve it as a next step. Just so it's more consistent across the components.

Reply:
This is fixed now